tests/api/commerce_decoupled_checkout/GiftCest.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+namespace falcon_commerce;
+
+use Codeception\Util\HttpCode;
+use Drupal\commerce_price\Price;
+
+/**
+ * Class GiftCest.
+ *
+ * Gift API tests.
+ *
+ * @package Falcon Commerce
+ */
+class GiftCest {
+
+  private $variation;
+
+  private $product;
+
+  /**
+   * @var array
+   */
+  private $post = [
+    'order' => [
+      'type' => 'gift',
+      'order_items' => [
+        [
+          'type' => 'gift',
+          'unit_price' => [
+            'number' => 15,
+            'currency_code' => 'USD',
+          ],
+          'purchased_entity' => [
+            'sku' => 'test gift',
+          ],
+        ],
+      ],
+    ],
+    'profile' => [
+      'field_phone' => '88001234567',
+      'field_contact_email' => 'allowed',
+      'field_contact_phone' => 'allowed',
+      'field_contact_sms' => 'denied',
+      'address' => [
+        'given_name' => 'Generous',
+        'family_name' => 'Donor',
+        'country_code' => 'US',
+        'address_line1' => '1098 Alta Ave',
+        'locality' => 'Mountain View',
+        'administrative_area' => 'CA',
+        'postal_code' => '94043',
+      ],
+    ],
+    'user' => [
+      'mail' => 'test+suite@systemseed.com',
+    ],
+    'payment' => [
+      'gateway' => 'example_test',
+      'type' => 'credit_card',
+      'details' => [
+        'type' => 'visa',
+        'number' => '4111111111111111',
+        'expiration' => [
+          'month' => '01',
+          'year' => '2022',
+        ],
+      ],
+    ],
+  ];
+
+  /**
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  public function _before() {
+    $entity_type_manager = \Drupal::entityTypeManager();
+
+    $stores = $entity_type_manager->getStorage('commerce_store')->loadByProperties(['type' => 'online']);
+
+    // Create product variation gift.
+    $this->variation = $entity_type_manager->getStorage('commerce_product_variation')->create([
+      'type' => 'gift',
+      'title' => 'Test gift',
+      'sku' => 'test gift',
+      'status' => 1,
+      'price' => new Price('0', 'USD'),
```

Comment:
Why does it have zero price?

Reply:
i changed to 10, but i think it doesn`t metter for this case.